cmd/nameset-backfill/main.go

```diff
@@ -0,0 +1,148 @@
+package main
+
+import (
+	"crypto/sha256"
+	"crypto/x509"
+	"flag"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/letsencrypt/boulder/cmd"
+	"github.com/letsencrypt/boulder/core"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/rpc"
+	"github.com/letsencrypt/boulder/sa"
+
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/cactus/go-statsd-client/statsd"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/gopkg.in/gorp.v1"
+)
+
+type resultHolder struct {
+	Serial  string
+	Expires time.Time
+	DER     []byte
+}
+
+type backfiller struct {
+	sa    core.StorageAuthority
+	dbMap *gorp.DbMap
+	stats statsd.Statter
+	log   *blog.AuditLogger
+}
+
+func new(amqpConf *cmd.AMQPConfig, syslogConf cmd.SyslogConfig, statsdURI, dbURI string) (*backfiller, error) {
+	var stats statsd.Statter
+	var err error
+	stats, log := cmd.StatsAndLogging(cmd.StatsdConfig{Server: statsdURI, Prefix: "Boulder"}, syslogConf)
+	sac, err := rpc.NewStorageAuthorityClient("nameset-backfiller", amqpConf, stats)
+	if err != nil {
+		return nil, err
+	}
+	dbMap, err := sa.NewDbMap(dbURI)
+	if err != nil {
+		return nil, err
+	}
+	return &backfiller{sac, dbMap, stats, log}, nil
+}
+
+func (b *backfiller) run() error {
+	added := 0
+	defer b.log.Info(fmt.Sprintf("Added %d missing certificate name sets to the nameSets table", added))
+	for {
+		results, err := b.findCerts()
+		if err != nil {
+			return err
+		}
+		if len(results) == 0 {
+			break
+		}
+		err = b.processResults(results)
+		if err != nil {
+			return err
+		}
+		added += len(results)
+	}
+	return nil
+}
+
+func (b *backfiller) findCerts() ([]resultHolder, error) {
+	var results []resultHolder
+	_, err := b.dbMap.Select(
+		&results,
+		// idk left outer join instead?
```

Comment:
left join is going to be faster, as it will be a progressive scan, but I'm not sure it really matters given that this is a single-shot util.

Still, the way you'd do that is something like `LEFT JOIN nameSets AS ns ON ns.serial=c.serial ... WHERE ns.serial IS NULL`